document/js-api/index.bs

```diff
@@ -679,36 +701,97 @@ which can be simultaneously referenced by multiple {{Instance}} objects. Each
-    1. [=Reset the memory buffer=] of |memaddr|.
+    1. [=Refresh the memory buffer=] of |memaddr|.
     1. Return |ret|.
 </div>
 
+<div algorithm=dom-Memory-grow>
+    The <dfn method for="Memory">grow(|delta|)</dfn> method, when invoked, performs the following steps:
+    1. Let |memaddr| be **this**.\[[Memory]].
+    1. Return the result of [=grow the memory buffer|growing the memory buffer=] associated with |memaddr| by |delta|.
+</div>
+
 Immediately after a WebAssembly [=memory.grow=] instruction executes, perform the following steps:
 
 <div algorithm="memory.grow">
     1. If the top of the stack is not [=i32.const=] (−1),
         1. Let |frame| be the [=current frame=].
         1. Assert: due to validation, |frame|.[=frame/module=].[=moduleinst/memaddrs=][0] exists.
         1. Let |memaddr| be the memory address |frame|.[=frame/module=].[=moduleinst/memaddrs=][0].
-        1. [=Reset the memory buffer=] of |memaddr|.
+        1. [=Refresh the memory buffer=] of |memaddr|.
+</div>
+
+<div algorithm=dom-Memory-toFixedLengthBuffer>
+    The <dfn method for="Memory">toFixedLengthBuffer()</dfn> method, when invoked, performs the following steps:
+    1. Let |buffer| be **this**.\[[BufferObject]].
+    1. If [=IsResizableArrayBuffer=](|buffer|) is false, return |buffer|.
+    1. Let |memaddr| be **this**.\[[Memory]].
+    1. Let |fixedBuffer| be the result of [=create a fixed length memory buffer|creating a fixed length memory buffer=] from |memaddr|.
+    1. Perform ! [=DetachArrayBuffer=](|buffer|, "WebAssembly.Memory").
+    1. Set **this**.\[[BufferObject]] to |fixedBuffer|.
+    1. Return |fixedBuffer|.
+</div>
+
+<div algorithm=dom-Memory-toResizableBuffer>
+    The <dfn method for="Memory">toResizableBuffer()</dfn> method, when invoked, performs the following steps:
+    1. Let |buffer| be **this**.\[[BufferObject]].
+    1. If [=IsResizableArrayBuffer=](|buffer|) is true, return |buffer|.
+    1. Let |memaddr| be **this**.\[[Memory]].
+    1. Let |store| be the [=surrounding agent=]'s [=associated store=].
+    1. Let |memtype| be [=mem_type=](|store|, |memaddr|).
+    1. If |memtype| has a max,
+        1. Let |maxsize| be the max value in |memtype|.
+    1. Otherwise,
+        1. Let |maxsize| be an implementation-defined value &ge; |buffer|.\[[ArrayBufferByteLength]] and &lt; (65536 &times; 65536).
+
+            Note: The implementation-defined value above is the same across all invocations of this method.
+
+    1. Let |resizableBuffer| be the result of [=create a resizable memory buffer|creating a resizable memory buffer=] from |memaddr| and |maxsize|.
+    1. Perform ! [=DetachArrayBuffer=](|buffer|, "WebAssembly.Memory").
+    1. Set **this**.\[[BufferObject]] to |resizableBuffer|.
+    1. Return |resizableBuffer|.
+</div>
+
+{{ResizableArrayBuffer}} objects returned by a {{Memory}} object must use the following definition of [=HostResizeArrayBuffer=].
```

Comment:
That's an interesting idea for hosts to write a method. It'll be a new editorial convention.

Do I understand the nit correctly that you see this as an implementation hook instead of a host hook, thus the suggestion?